src/main/kotlin/eu/europa/ec/eudi/pidissuer/adapter/out/diploma/IssueJwtVcJsonDiploma.kt

```diff
@@ -0,0 +1,188 @@
+    id = CredentialConfigurationId((DiplomaJwtVcJsonScope.value)),
+    scope = DiplomaJwtVcJsonScope,
+    cryptographicBindingMethodsSupported = emptySet(),
+    credentialSigningAlgorithmsSupported = emptySet(),
+    display = diplomaDisplay,
+    proofTypesSupported = nonEmptySetOf(ProofType.Jwt(nonEmptySetOf(JWSAlgorithm.RS256, JWSAlgorithm.ES256))),
+    credentialDefinition = CredentialDefinition(
+        type = listOf("VerifiableCredential", "UniversityDegreeCredential"),
+        credentialSubject = listOf(
+            AchievedClaim.attribute,
+            EntitledToClaim.attribute,
+            PerformedClaim.attribute,
+        ),
+    ),
+)
+
+class IssueJwtVcJsonDiploma(
+    val credentialIssuerId: CredentialIssuerId,
+    private val generateUserDiplomaData: GenerateUserDiplomaData,
+    private val notificationsEnabled: Boolean,
+    private val generateNotificationId: GenerateNotificationId,
+    private val extractJwkFromCredentialKey: ExtractJwkFromCredentialKey,
+    private val issuerSigningKey: IssuerSigningKey,
+    private val clock: Clock,
+    private val storeIssuedCredential: StoreIssuedCredential,
+) : IssueSpecificCredential<JsonElement> {
+
+    private val log = LoggerFactory.getLogger(IssueJwtVcJsonDiploma::class.java)
+
+    override val supportedCredential: JwtVcJsonCredentialConfiguration
+        get() = IssueJwtVcJsonDiplomaConfiguration
+
+    override val publicKey: JWK? = null
+
+    private val validateProof = ValidateProof(credentialIssuerId)
+
+    context(Raise<IssueCredentialError>)
+    override suspend fun invoke(
+        authorizationContext: AuthorizationContext,
+        request: CredentialRequest,
+        credentialIdentifier: CredentialIdentifier?,
+        expectedCNonce: CNonce,
+    ): CredentialResponse<JsonElement> = coroutineScope {
+        log.info("Handling issuance request ...")
+        val holderPubKey = async(Dispatchers.Default) { holderPubKey(request, expectedCNonce) }
+
+        val jwt = encodeDiplomaToJwtVcJson(holderPubKey.await())
+
+        val notificationId =
+            if (notificationsEnabled) generateNotificationId()
+            else null
+
+        storeIssuedCredential(
+            IssuedCredential(
+                format = JWT_VC_JSON_FORMAT,
+                type = supportedCredential.credentialDefinition.type.joinToString(" "),
+                holder = holderPubKey.await().toPublicJWK().computeThumbprint().toString(),
+                holderPublicKey = holderPubKey.await().toPublicJWK(),
+                issuedAt = clock.instant(),
+                notificationId = notificationId,
+            ),
+        )
+
+        CredentialResponse.Issued(JsonPrimitive(jwt), notificationId)
+            .also {
+                log.info("Successfully issued diploma")
+                log.debug("Diploma data {}", it)
+            }
+    }
+
+    private fun encodeDiplomaToJwtVcJson(holderKey: JWK): String {
```

Comment:
Are we sure that holderPubKey shouldn't be a `JWK`?

It should be a DID or DIDURL, I think.